tensorflow/core/kernels/functional_ops.cc

```diff
@@ -401,12 +399,14 @@ class WhileOp : public AsyncOpKernel {
   static Status CondResultToBool(OpKernelContext* ctx,
                                  const FunctionLibraryRuntime::Options& opts,
                                  const Tensor& cond_t, bool* out_result) {
-#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
+    bool is_pluggable = ctx->op_device_context()
+                            ? ctx->op_device_context()->IsPluggableDevice()
+                            : false;
```

Comment:
We can simplify this to
```suggestion
    bool is_pluggable = ctx->op_device_context() &&
                        ctx->op_device_context()->IsPluggableDevice();
```